libs/langchain/langchain/vectorstores/neo4j_vector.py

```diff
@@ -0,0 +1,623 @@
+            (default: False). Useful for testing.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.vectorstores.neo4j_vector import Neo4jVector
+            from langchain.embeddings.openai import OpenAIEmbeddings
+
+            url="bolt://localhost:7687"
+            username="neo4j"
+            password="pleaseletmein"
+            embeddings = OpenAIEmbeddings()
+            vectorestore = Neo4jVector.from_documents(
+                embedding=embeddings,
+                documents=docs,
+                url=url
+                username=username,
+                password=password,
+            )
+
+
+    """
+
+    def __init__(
+        self,
+        username: str,
+        password: str,
+        url: str,
+        embedding_function: Embeddings,
+        database: str = "neo4j",
+        index_name: str = "vector",
+        node_label: str = "Chunk",
+        embedding_node_property: str = "embedding",
+        text_node_property: str = "text",
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+        logger: Optional[logging.Logger] = None,
+        pre_delete_collection: bool = False,
+        relevance_score_fn: Optional[Callable[[float], float]] = None,
+    ) -> None:
+        try:
+            import neo4j
+        except ImportError:
+            raise ValueError(
+                "Could not import neo4j python package. "
+                "Please install it with `pip install neo4j`."
+            )
+
+        self._driver = neo4j.GraphDatabase.driver(url, auth=(username, password))
+        self._database = database
+        self.schema = ""
+        # Verify connection
+        try:
+            self._driver.verify_connectivity()
+        except neo4j.exceptions.ServiceUnavailable:
+            raise ValueError(
+                "Could not connect to Neo4j database. "
+                "Please ensure that the url is correct"
+            )
+        except neo4j.exceptions.AuthError:
+            raise ValueError(
+                "Could not connect to Neo4j database. "
+                "Please ensure that the username and password are correct"
+            )
+
+        # Verify if the version support vector index
+        self.verify_version()
+
+        self.embedding_function = embedding_function
+        self._distance_strategy = distance_strategy.value
+        self.index_name = index_name
+        self.node_label = node_label
+        self.embedding_node_property = embedding_node_property
+        self.text_node_property = text_node_property
+        self.logger = logger or logging.getLogger(__name__)
+        self.override_relevance_score_fn = relevance_score_fn
+
+        # Calculate embedding dimension
+        self.embedding_dimension = len(embedding_function.embed_query("foo"))
+
+        # Delete existing data if flagged
+        if pre_delete_collection:
+            self.query(f"MATCH (n:{self.node_label}) DETACH DELETE n")
```

Comment:
we should batch this:

MATCH (n:{self.node_label}) 
CALL { WITH n DETACH DELETE n } IN TRANSACTIONS OF 10000 ROWS;

Reply:
Still dangerous if someone switches that on :)

Reply:
wrap {self.node_label} in backticks